int32 ptNum
tunneloutpoint[] top_arr